qiling/__init__.py

```diff
@@ -1,2 +1,78 @@
 from .core import Qiling
-from .__version__ import __version__
\ No newline at end of file
+from .__version__ import __version__
+
+def __git_info():
+    """Retrieve git information of the current Qiling code base.
+    """
+
+    import os
+
+    gitdir = os.path.join(os.path.dirname(os.path.abspath(__file__)), r'..', r'.git')
+    head_path = 'HEAD'
+
+    try:
+        with open(os.path.join(gitdir, head_path), 'r') as head_file:
+            head_info = head_file.readline().rstrip()
+
+        branch_path = os.path.join(gitdir, head_info.split('ref:', 1)[-1].lstrip())
+
+        with open(os.path.join(gitdir, branch_path), 'r') as branch_file:
+            commit = branch_file.readline().rstrip()
+
+        branch = os.path.basename(branch_path)
+    except OSError:
+        branch = None
+        commit = None
+
+    return {
+        'branch' : branch or 'unknown',
+        'commit' : commit or 'unknown'
+    }
+
+def __platform_info():
+    """Retrieve hosting platform information.
+    """
+
+    import platform
+
+    return {
+        'system'  : platform.system(),
+        'release' : platform.release(),
+        'machine' : platform.machine(),
+        'python'  : platform.python_version()
+    }
+
+def __libs_info(modules):
+    """Retrieve Python dependencies information.
+    """
+
+    def __module_ver(mname: str):
+        """Query a module's version by name.
+        """
+
+        import importlib
+
+        try:
+            m = importlib.import_module(mname)
```

Comment:
A better approach is `importlib.metadata.version` I think.

Reply:
Nice, I didn't know this one..
It seems to pull the version number from `pip` rather than relying on the `__version__` convension. In case of unicorn, it shows `'2.0.0rc4'` instead of just `2.0.0`

Reply:
Yes, that's exactly what I expect.